src/plugins/ds_mongo.c

```diff
@@ -0,0 +1,3126 @@
+/**
+ * @file ds_mongo.c
+ * @author Ondrej Kusnirik <Ondrej.Kusnirik@cesnet.cz>
+ * @brief internal MongoDB datastore plugin
+ *
+ * @copyright
+ * Copyright (c) 2021 - 2022 Deutsche Telekom AG.
+ * Copyright (c) 2021 - 2022 CESNET, z.s.p.o.
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define _GNU_SOURCE
+
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <grp.h>
+#include <pthread.h>
+#include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#include <libyang/libyang.h>
+
+#include "ds_mongo_config.h"
+#include "plugins_datastore.h"
+#include "sysrepo.h"
+
+#define plugin_name "MONGO DS"
+
+#define WRAPERR(err, type, func, message) srplg_log_errinfo(err, plugin_name, NULL, type, func " failed on %d in " __FILE__ " [%s].", __LINE__, message);
+
+typedef struct mongo_data_s {
+    /* client that connects to the database and
+            manages changes on the data */
+    mongoc_client_t *client;
```

Comment:
Doxygen comments for struct members are written:
```
type var;  /**< this is a var */
```

Reply:
done